### ISSUE TITLE
Not able to update resource context

### ISSUE THREAD

**Issue by theKashey** [atlassian-labs/react-resource-router#37]
As I understand resourceContext might be used to fed `getKey`/`getData` with extra props, not reflected in the `query`. However the updates to it are ignored, and context is never updated.

Ref: https://atlassian-labs.github.io/react-resource-router/#/router/configuration?id=resourcecontext

If context is not supposed to be used in that way - then what can I do to provide _extra variables_ to my resources? For example language - it might be not a part of URL, but might the changed in runtime.

**Comment by albertogasparin** [comment]
Yes, I think this is probably a bug. We never hit that case as we need full page reload to change language or user. But should handle it properly
